packages/datadog-esbuild/src/utils.js

```diff
@@ -7,9 +7,22 @@ const path = require('node:path')
 const { NODE_MAJOR, NODE_MINOR } = require('../../../version.js')
 
 const getExportsImporting = (url) => import(url).then(Object.keys)
-const getExports = NODE_MAJOR >= 20 || (NODE_MAJOR === 18 && NODE_MINOR >= 19)
-  ? require('import-in-the-middle/lib/get-exports.js')
-  : getExportsImporting
+let getExportsModulePromise
+
+const loadGetExportsModule = () => {
+  if (!getExportsModulePromise) {
+    getExportsModulePromise = import('import-in-the-middle/lib/get-exports.mjs')
+  }
+  return getExportsModulePromise
+}
+
+const getExports = async (srcUrl, context, getSource) => {
+  if (NODE_MAJOR >= 20 || (NODE_MAJOR === 18 && NODE_MINOR >= 19)) {
+    const mod = await loadGetExportsModule()
+    return mod.getExports(srcUrl, context, getSource)
+  }
+  return getExportsImporting(srcUrl, context, getSource)
+}
```

Comment:
This does a runtime check each time the method is called. Before, it would create the proper method depending on the version, never doing the runtime check.

Could you please change that back to the former behavior? :) 

```suggestion
const getExports = NODE_MAJOR >= 20 || (NODE_MAJOR === 18 && NODE_MINOR >= 19
  ? async (srcUrl, context, getSource) => {
    const mod = await loadGetExportsModule()
    return mod.getExports(srcUrl, context, getSource)
  }
  : getExportsImporting
```